### DISCUSSION TITLE
SelectColumn to support nullable()

### DISCUSSION THREAD

**Discussion by ziming** [filamentphp/filament#5446] [Ideas]
Currently SelectColumn does not support null as a choice value, however database columns being nullable is a common thing.

It will be great if SelectColumn can support nullable() just like how the form Select field support it.

**Comment by danharrin** [comment]
You can PR a fix if you want!

**Reply by ziming** [reply]
I figured out an alternative with column action so I will use that instead, thanks!

**Reply by khanakia** [reply]
@ziming how did you do that ?

**Reply by ziming** [reply]
action on a column then form

**Reply by erwindeclerck** [reply]
Hi
I encountered the same problem...
@ziming could you be a little bit more specific on how you managed this 'action on a column then form' solution?
Kind regards,
Erwin

**Reply by khanakia** [reply]
make your column fillable in Model file to solve this

**Reply by erwindeclerck** [reply]
Hi Khanakia

Thanks for your response.
I'm afraid I don't see it immediately... 
But perhaps it helps if I explain what I'm trying to do:
For a an exchange-program for professionals in youth-carers have a list of 'hosts' that offer (a limited number of) visitation opportunities to 'guests'.
The guests can postulate for (max) three hosts and enter their priority list in a form in three fields with foreign keys to the hosts of their choice.
I then have a table with corresponding Model (matches) in which an administrator attributes what he or she thinks is the best match.
I create a drop-down-list with the (max) three choices in order of their preference.
At the start the match-column is empty, and as soon as a choice has been made, the field is nicely saved in the matching-table.
But if they change their mind, they can not undo their choice by 'unsetting' the dropdown, because then the database gets an update with an empty, not a NULL... and the field is an int 
I offer this dropdown in a 'table-form' because it is important to give the administrator as much context as possible about other guests and hosts. If not I would do it in the individual form, but that would be too much distracting.

This is my Select-column:


```
                SelectColumn::make('host')
                    ->options(function (Matching $record){
                        $options[1985] = "Make a match";//attempt to create a fake host...
                        $choice1 = $record->guest()->get()->first()->host1()->get()->first();
                        $saldi = $choice1->saldi()->first();
                        $options[$choice1->id] = "1 -> " . $choice1->full_name ." ( M:". $saldi->max ." A:".$saldi->assigned ." R:".$saldi->remainder ." )";
                        $choice2 = $record->guest()->get()->first()->host2()->get()->first();
                        $saldi = $choice2->saldi()->first();
                        $options[$choice2->id] = "2 -> " . $choice2->full_name ." ( M:". $saldi->max ." A:".$saldi->assigned ." R:".$saldi->remainder ." )";
                        $choice3 = $record->guest()->get()->first()->host3()->get()->first();
                        $saldi = $choice3->saldi()->first();
                        $options[$choice3->id] = "3 -> " . $choice3->fullname ." ( M:". $saldi->max ." A:".$saldi->assigned ." R:".$saldi->remainder ." )";
                        return $options ;
                    })->disabled(function(Matching $record){
                        return $record->lock;
                    })->default(function(){return 1985;})//second attempt to offer the fake host...
                    ->rules(['required']), //to avoid null being sent
```

I add as much info to the administrator to quickly see the context of the three opportunities...

As you suggested, the host-field is fillable, so that didn't help me further...

Would there be a way to override the store-action of this component so that I can write my own query to update this field in the db-table and set a NULL...?

Erwin

**Reply by khanakia** [reply]
yes, that's true on unsetting the dropdown the DB column will be set as EMPTY not NULL but I was ok with that.

I believe you can use `mutateFormDataBeforeSave` to override the value and set it to NULL before saving into the DB https://filamentphp.com/docs/3.x/panels/resources/editing-records#customizing-data-before-saving

**Reply by erwindeclerck** [reply]
Thx for the suggestion.
Tried it out but don't seem to be able to intercept the save-action when working from inside the SelectColumn (= table field)
It works if I work via the from. But that's no option for my use case.
The alternative for handling forms via modals doesn't work either in the table context...
https://filamentphp.com/docs/3.x/actions/prebuilt-actions/edit#customizing-the-saving-process
Hooks don't work either because they create an edit link...
Any other ideas?
;-)

**Reply by danharrin** [reply]
Sounds like a bug, so is there an issue submitted with a reproduction repository?
